pilot/cmd/pilot-agent/main.go

```diff
@@ -150,25 +150,27 @@ var (
 			role.TrustDomain = spiffe.GetTrustDomain()
 			log.Infof("Proxy role: %#v", role)
 
+			proxyConfig := model.DefaultProxyConfig()
+
 			// Add cert paths as node metadata only if they differ from defaults
 			if tlsServerCertChain != model.DefaultCertChain {
-				role.Metadata[model.NodeMetadataTLSServerCertChain] = tlsServerCertChain
+				proxyConfig.EnvoyNodeMetadata[model.NodeMetadataTLSServerCertChain] = tlsServerCertChain
```

Comment:
The `role.Metadata` was never initialized or read, so setting it was a fail. Instead we are using the new `EnvoyNodeMetadata` field on the ProxyConfig struct. This gets passed in to the bootstrap code so it seems like the logical place for it to live.

Reply:
instead of piggybacking on metadata, cant you do this check after https://github.com/istio/istio/pull/13367/files#diff-5cb6a377bcea0cba4f2dd118a326af06R181 has been called?

Reply:
@rshriram I'm not sure what you mean by "piggybacking on metadata". The cert file overrides are used by both the file watcher here in pilot-agent and they are sent as node metadata so they can be used by pilot to configure listeners + clusters.